include/pybind11/options.h

```diff
@@ -17,32 +17,30 @@ class options {
 public:
 
     // Default RAII constructor, which leaves settings as they currently are.
-    options() : previous_state(global_state()) {}
+    options();
```

Comment:
Still an interesting discussion to be had: where do we draw the line. I.e., even in "normal" C++ code, split up as usual over headers and cpp-files, this constructor might have been declared here.

I guess this won't be the part adding compilation overhead?

Reply:
Same with the other functions, actually. This might be a part that's good to keep as is, and to have inline even in a build with source files.

Reply:
In this particular case, it is not immediately obvious if the function is trivial or not, because `global_state()` could in theory expand to a huge inlined function (although this is unlikely since large functions are generally not inlined). A better rule might be to keep in headers only constructor that perform only trivial assignments.